### ISSUE TITLE
Approximate interaction values

### ISSUE THREAD

**Issue by alexcoca** [shap/shap#1171]
Hi @slundberg ,

I was looking at the code and was wondering why Sabaas values are not supported for LightGBM and Catboost models. On the surface of it, it looks as though removing the assertions could see the algorithm use the C extension for computing the values? 

I looked a bit deeper and saw that for LightGBM the above might not be possible because the "C extension cannot handle categorical splits yet". What doest this mean? That we can't pass data matrices with string values for different levels of categorical variables? I expect that passing an array where categorical variables are encoded numerically (e.g., 0.0, 1.0, 2.0, ...) would work. So it seems that currently the solution is to use the LightGBM implementation directly as opposed to the C extension just so the case where the user passes a mixed types (string and float) data matrix is supported? The same seems to apply to Catboost.

So, to summarise, we could, in theory, compute the Sabaas values for Catboos and LightGBM if the user encodes their feature matrix or if all their features are continuos, if the assertions were removed and the control flow would be updated? As things stand, the approximate values can't be computed and the TreeShap algorithm runs from Catboost/LightGBM.

**Comment by Tachyon5** [comment]
a bit disturbing to see nobody has answered this.  I resolved this issue by doing what you said; encoding the categorical variables as numeric using WOE/Mean Encoding.
